src/edu/sjsu/stratagem/ExpressionBuilderVisitor.java

```diff
@@ -152,15 +161,31 @@ public Expression visitUnit(StratagemParser.UnitContext ctx) {
 		return new ValueExpr(UnitVal.singleton);
 	}
 
-	@Override
-	public Expression visitId(StratagemParser.IdContext ctx) {
-		String id = ctx.ID().getText();
-		return new VarExpr(id);
+	private Type parseType(StratagemParser.TypeContext ctx) {
+		if (ctx.type_prim() != null) {
+			return parsePrimitiveType(ctx.type_prim());
+		} else {
+			return parseClosureType(ctx.type_fun());
+		}
 	}
 
-	@Override
-	public Expression visitPrint(StratagemParser.PrintContext ctx) {
-		Expression arg = visit(ctx.args().getChild(1));
-	    return new PrintExpr(arg);
+	private Type parsePrimitiveType(StratagemParser.Type_primContext ctx) {
+		if (ctx.TYPE_BOOL() != null) {
+			return BoolType.singleton;
+		} else if (ctx.TYPE_INT() != null) {
+			return IntType.singleton;
+		} else if (ctx.TYPE_STRING() != null) {
+			return StringType.singleton;
+		} else if (ctx.UNIT() != null) {
+			return UnitType.singleton;
+		} else {
+			throw new StratagemException("Unknown primitive type");
```

Comment:
It might make sense to have specific exceptions based on the error type, so StrategemRuntimeException, StratagemTypecheckException, etc.